horovod/common/operations.cc

```diff
@@ -848,6 +848,9 @@ void BackgroundThreadLoop(HorovodGlobalState& state, MPIContext& ctx) {
   // Signal that shutdown has been requested.
   state.shut_down = true;
 
+#if HAVE_NCCL && HOROVOD_GPU_ALLREDUCE == 'N'
```

Comment:
Since we create the `NCCLContext` regardless of `HOROVOD_GPU_ALLREDUCE` value, I think this should just be:

```
if HAVE_NCCL`
```